aws-codeartifact-repository/src/main/java/software/amazon/codeartifact/repository/Translator.java

```diff
@@ -128,18 +129,22 @@ static DescribeRepositoryRequest translateToReadRequest(final ResourceModel mode
    */
   static ResourceModel translateFromReadResponse(final DescribeRepositoryResponse describeRepositoryResponse) {
     RepositoryDescription repositoryDescription = describeRepositoryResponse.repository();
-    return ResourceModel.builder()
-        // TODO add external connections and policy document
-        // https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-codeartifact/issues/18
+    ResourceModel resourceModelBuilder = ResourceModel.builder()
+        // TODO add repositoryEndpoint
         .arn(repositoryDescription.arn())
         .description(repositoryDescription.description())
         .administratorAccount(repositoryDescription.administratorAccount())
         .domainName(repositoryDescription.domainName())
-        .upstreams(translateToUpstreamsFromRepoDescription(describeRepositoryResponse.repository()))
         .domainOwner(repositoryDescription.domainOwner())
         .description(repositoryDescription.description())
         .repositoryName(repositoryDescription.name())
         .build();
+
+    if (!CollectionUtils.isEmpty(repositoryDescription.upstreams())) {
+      resourceModelBuilder.setUpstreams(translateToUpstreamsFromRepoDescription(describeRepositoryResponse.repository()));
+    }
```

Comment:
not sure how the builder here works (if it allows you to set things after the creation), but if this is like the immutables library, it will not be happy with this since the ResourceModel is already constructed. You would have to not call `build()` yet, conditionally add the upstream, then return the `builder.build()`

Not sure you actually have to do this if this works, this is just how I would've written it.

Reply:
yes I agree with you, I meant to have done this. I'll update it